### PR TITLE
Dynamic configuration

### PR DESCRIPTION
**Pull request by carlk3**
Move hw_config.c out of FatFs_SPI library so that the library can better function as a submodule.

Add dynamic configuration example in directory dynamic_config_example.

### CODE REVIEW

File: simple_example/hw_config.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    } else {
+        return NULL;
+    }
+}
+FATFS *get_fs_by_name(const char *name) {
+    for (size_t i = 0; i < sd_get_num(); ++i)
+        if (0 == strcmp(sd_cards[i].pcName, name)) return &sd_cards[i].fatfs;
+    DBG_PRINTF("%s: unknown name %s\n", __func__, name);
+    return NULL;
+}
```
Comment (fruit-bat) [inline simple_example/hw_config.c:95]: Possibly redundant? Perhaps get_fs_by_name should be removed from hw_config.h?

File: simple_example/hw_config.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    } else {
+        return NULL;
+    }
+}
+sd_card_t *sd_get_by_name(const char *const name) {
+    for (size_t i = 0; i < sd_get_num(); ++i)
+        if (0 == strcmp(sd_cards[i].pcName, name)) return &sd_cards[i];
+    DBG_PRINTF("%s: unknown name %s\n", __func__, name);
+    return NULL;
+}
```
Comment (fruit-bat) [inline simple_example/hw_config.c:81]: Possibly redundant? Perhaps sd_get_by_name should be removed from hw_config.h?

File: dynamic_config_example/hw_config.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    } else {
+        return NULL;
+    }
+}
+sd_card_t *sd_get_by_name(const char *name) {
+    for (size_t i = 0; i < sd_get_num(); ++i)
+        if (0 == strcmp(sd_cards[i]->pcName, name)) return sd_cards[i];
+    DBG_PRINTF("%s: unknown name %s\n", __func__, name);
+    return NULL;
+}
```
Comment (fruit-bat) [inline dynamic_config_example/hw_config.cpp:32]: Possibly redundant? Perhaps sd_get_by_name should be removed from hw_config.h?

File: dynamic_config_example/hw_config.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    } else {
+        return NULL;
+    }
+}
+FATFS *get_fs_by_name(const char *name) {
+    for (size_t i = 0; i < sd_get_num(); ++i)
+        if (0 == strcmp(sd_cards[i]->pcName, name)) return &sd_cards[i]->fatfs;
+    DBG_PRINTF("%s: unknown name %s\n", __func__, name);
+    return NULL;
+}
```
Comment (fruit-bat) [inline dynamic_config_example/hw_config.cpp:46]: Possibly redundant? Perhaps get_fs_by_name should be removed from hw_config.h?

**Review by fruit-bat — COMMENTED** [review]
I've tried the changes in my application and they work fine... many thanks! I've added a few comments which relate to a potential minor tidy up of the interface. I don't think sd_get_by_name() or get_fs_by_name() form a part of it... any more. Apologies, if this is my misunderstanding.

File: simple_example/hw_config.c (hunk 1 above)
Comment (carlk3) [reply simple_example/hw_config.c]: OK, I moved sd_get_by_name and get_fs_by_name out of hw_config.

Comment (fruit-bat) [reply simple_example/hw_config.c]: Cheers

**Pull request merged**